src/peer/negotiator.js

```diff
@@ -477,6 +438,77 @@ class Negotiator extends EventEmitter {
   }
 
   /**
+   * Replace the stream being sent with a new one.
+   * @param {MediaStream} newStream - The stream to replace the old stream with.
+   * @private
+   */
+  _replacePerTrack(newStream) {
+    const vTracks = newStream.getVideoTracks();
+    const aTracks = newStream.getAudioTracks();
+    let vSenders = [];
+    let aSenders = [];
+    this._pc.getSenders().forEach(sender => {
+      sender.track.kind === 'video' && vSenders.push(sender);
+      sender.track.kind === 'audio' && aSenders.push(sender);
+    });
+
+    // no video tracks in newStream => remove all RtpSender(.kind = video)
+    // otherwise, replace it or create it
+    // We assume that there is at most 1 audio stream and at most 1 video stream in localStreams
+    if (vTracks.length === 0) {
+      vSenders.forEach(sender => this._pc.removeTrack(sender));
+    } else {
+      vTracks.forEach((track, idx) => {
+        if (vSenders.length === 0) {
+          // no video track in previous stream, so we can't replace it
+          this._pc.addTrack(track, newStream);
+        } else {
+          // need to check id before replace..?
+          vSenders[idx].replaceTrack(track);
```

Comment:
As I commented here, what will happen when trying to replace track A with track A?